Added support for encrpyted files

peter-kwan opened the pull request:
## Summary (required)

1. Add support for uploading and downloading files in the encrypted partition in S3. The CLI would switch to use the /encrypted-files/ endpoints for Putting and Getting those files, instead of using pre-signed URLs. No multi-part support because we expect these files to be small, typically under 100KB.

2. Fixed eEgis tests.

Chariot backend has support for this in production.

## JIRA (required)

https://linear.app/praetorian/issue/PRA-1469/one-click-launch-ui-and-history-tracking

peter-kwan committed:
added support for encrpyted files

peter-kwan committed:
added Accept application/octet-stream in the header when downloading directly from the encrypy-file/ lambda endpoint. It is needed to force the AWS API Gateway to return things in binary

peter-kwan committed:
added change log entry

peter-kwan committed:
removed the Accept header because I figured out how to force the AWS Gateway to return binary data

peter-kwan committed:
Revert "removed the Accept header because I figured out how to force the AWS Gateway to return binary data"

This reverts commit 2ae2f26c00d032b6068bf972948a584ea6800262.

peter-kwan committed:
nit

EvanLeleux commented on praetorian_cli/sdk/chariot.py:
it looks like we are allowing headers to be passed in here to then be reassigned. maybe just change the request() method to merge headers into kwargs['headers'] then remove the headers=headers? Dont know if that works

peter-kwan replied:
Good idea. I read your comment this way -- chariot_request() and request() are sort of redundant. In the original code, request() takes care of proxy. and chariot_request() takes care of the beta flag and adding the authentication headers.

Why not just have request() does all of it. 

And this would simplify the handling of headers.

In fact, not one single original callers of chariot_request() sets the headers argument.

I will simplify it.

peter-kwan replied:
addressed here: de21d854

peter-kwan committed:
added tests for uploading to the encrypted partition

peter-kwan committed:
addressed review comment for consolidating Chariot.chariot_request() and Chariot.request()

peter-kwan committed:
fix tests

peter-kwan committed:
resolved merge conflicts

peter-kwan commented:
@EvanLeleux  this is ready to be reviewed again. I addressed your comment to simplify chariot_request() and request() in chariot.py. 

This also fixes a bunch of test failures that I got running the tests on `main`.

peter-kwan commented:
Also ready to be published next time around since the backend in prod supports the _encrypted/ partition now.

peter-kwan committed:
Merge branch 'main' into peter/encrypted-files

peter-kwan committed:
Update CHANGELOG.md
